keymanager/src/runtime/methods.rs

```diff
@@ -54,35 +57,84 @@ const MAX_EPHEMERAL_KEY_AGE: EpochTime = 10;
 /// of blocks lower than the height of the latest trust root.
 const MAX_FRESH_HEIGHT_AGE: u64 = 50;
 
-/// The size of an encrypted ephemeral secret.
-const EPHEMERAL_SECRET_STORAGE_SIZE: usize = 32 + TAG_SIZE + NONCE_SIZE;
-
 /// Initialize the Kdf.
 pub fn init_kdf(ctx: &mut RpcContext, req: &InitRequest) -> Result<SignedInitResponse> {
+    let rctx = runtime_context!(ctx, KmContext);
+    let runtime_id = rctx.runtime_id;
+    let storage = ctx.untrusted_local_storage;
+
     let policy_checksum = Policy::global().init(ctx, &req.policy)?;
-    Kdf::global().init(req, ctx, policy_checksum)
+
+    let kdf = Kdf::global();
+    kdf.set_runtime_id(runtime_id)?;
+
+    // Load master secrets if not already loaded. It is intended that manual intervention
+    // by the operator is required to remove/alter persisted secrets.
+    kdf.load_master_secrets(storage, &runtime_id)?;
+
```

Comment:
Eventually we should verify the request against consensus layer state.

Reply:
Ok, will do it in another PR after rotations.